data/service/data-operation.js

```diff
@@ -147,14 +341,65 @@ exports.DataOperation = Montage.specialize(/** @lends DataOperation.prototype */
 
 }, /** @lends DataOperation */ {
 
+
+    /* Explore the opportunity to add more XHR-like API with events matching the semantic of possible sytmetric operations
+        and also promise API that would resolve to the matching operation success/failed
+    */
+
     Type: {
         value: {
-            CREATE: {isCreate: true},
-            READ: {isRead: true},
-            UPDATE: {isUpdate: true},
-            DELETE: {isDelete: true}
+            Create: {isCreate: true},
+            CreateFailed: {isCreate: true},
+            CreateCompleted: {isCreate: true},
+            //Additional
+            Copy: {isCreate: true},
+            CopyFailed: {isCreate: true},
+            CopyCompleted: {isCreate: true},
+            /* Read is the first operation that mnodels a query */
+            Read: {isRead: true},
```

Comment:
Do we need a "ReadProperty" and parallel set of accompanying operations to model when we fetch individual properties of objects?

Reply:
That an operation is meant to read a property strikes me as a characteristic of a "Read" operation rather than a separate kind of operation altogether. If the flow for a read goes: 
1. Is this a read? Check `anOperation.type`. 
2. What are we reading? Check `anOperation.criteria`. 

Then we should store the information about "is this for a property" in the criteria, not the type. 

Or consider an `Update` operation.  `Update` doesn't say in and of itself whether a single property or every property is modified. The specifics about which properties are modified are passed with the data and criteria. It is not several `UpdateProperty` operations.

Reply:
👍🏻

Reply:
TODO: It's missing authentication related API